### ISSUE TITLE
Accuracy and code related question!

### ISSUE THREAD

**Issue by MitalPattani** [canqin001/PointDAN#11]
Question1: I have read your paper and been through your code, can you explain the code you have used to get the class-level accuracy that you have mentioned in your paper? It would be great if you could add that bit to the code base!
Question2: For example, modelnet dataset has size 4183 and shapenet has size 17,378 , while training you are only utilizing the first 4183 examples from both datasets while training S→M have you considered randomly initializing the training examples from shapenet (larger dataset). I have a point cloud dataset with CAD and CAM images where my CAD training dataset is almost 6 times larger than CAM training dataset and I am considering randomly initializing the subset of dataset used for after 100 epochs, what do you think?

**Comment by canqin001** [comment]
Hi,

Thanks for your interest of our work. I am a little confused about Q1. Which part of code is confusing for you? For Q2, we suggest you take all the data for training, which is even imbalanced. But you have to make sure that the class-space are totally overlapped.

Thanks,
Can

**Comment by MitalPattani** [comment]
Hi,
Thank you for your response,
Question1: I am talking about Table 3 in your paper, can you upload the code you have used to make this class-wise classification table.
http://papers.nips.cc/paper/8940-pointdan-a-multi-scale-3d-domain-adaption-network-for-point-cloud-representation.pdf
Question2 : I am considering simply duplicating the target dataset (smaller dataset) 6 times, so as to avoid imbalance, and utilize the entire dataset. If I am not wrong you are not utilizing the entire 17,378 when there is an imbalance, correct?

**Comment by canqin001** [comment]
For Q1, the codes in line 188-193 in train_source.py are used for class-wise acc. For Q2, we have tried both actually. The results reported in paper are imbalance one. According to our results, we find that more data are useful for improving the performance. But it depends on the cases. If not taking a lot of time, I suggest you try both to compare the difference.

**Comment by MitalPattani** [comment]
Alright, thanks! :)

**Comment by MitalPattani** [comment]
Hey,
Sorry for disturbing you again, but the code is not using imbalanced data it seems! I tried to replicate the results and I checked the count!

**Comment by Hxyou** [comment]
@MitalPattani If my recall is right, we are using all the data in our released code. But your idea of resampling to make the data more balanced is worth trying. There are several commonly used resampling methods or you can even try with a class reweighting loss. 

Best,
Haoxuan

**Comment by MitalPattani** [comment]
alright, I've put it to train let's see, thanks!
